src/types/enums.rs

```diff
@@ -507,3 +548,21 @@ impl<'ctx> From<BasicTypeEnum<'ctx>> for BasicMetadataTypeEnum<'ctx> {
         }
     }
 }
+
+impl Display for AnyTypeEnum<'_> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}", self.print_to_string())
+    }
+}
+
+impl Display for BasicTypeEnum<'_> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}", self.print_to_string())
+    }
+}
+
+impl Display for BasicMetadataTypeEnum<'_> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}", self.print_to_string())
```

Comment:
These should probably make mention of the wrapping type, ie `"BasicMetadataTypeEnum::{}", self.print_to_string())` or something like that?

Reply:
I think that should the task of the `Debug` trait, by using the marker `"{:?}"`.

For the `Display` trait, the Type names should be printed like how LLVM Types are printed in C++ API.

How do you think about that?

Reply:
Hmm I suppose that's fair